Shipr-Helm/templates/redis-deployment.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: apps/v1
+kind: {{  .Values.redis.kind }}
+metadata:
+  name: {{  .Values.redis.name }}
+spec:
+  replicas: {{  .Values.redis.replicas }}
+  selector:
+    matchLabels:
+      app: {{  .Values.redis.name }}
+  template:
+    metadata:
+      labels:
+        {{- with .Values.redis.labels }}
+        {{- toYaml . | nindent 6 }}
+        {{- end }}
+    spec:
+      containers:
+      - name: {{  .Values.redis.name }}
+        image: {{  .Values.redis.repository }}:{{  .Values.redis.tag }}
+        ports:
+        - containerPort: {{  .Values.redis.containerPort }}
+        {{- if .Values.redis.probes.liveness.enabled }}
+        livenessProbe:
+          tcpSocket:
+            port: {{  .Values.redis.containerPort }}
+          initialDelaySeconds: 10
+          periodSeconds: 10
+        {{- end }}
+        {{- if .Values.redis.probes.readiness.enabled }}
+        readinessProbe:
+          tcpSocket:
+            port: {{  .Values.redis.containerPort }}
+          initialDelaySeconds: 5
+          periodSeconds: 5
+        {{- end }}
+        volumeMounts:
+        - mountPath: /data
+          name: redis-storage
+      volumes:
+      - name: redis-storage
+        persistentVolumeClaim:
+          claimName: redis-pvc
```

Comment:
Since we have allowed statefulset too, there should be `volumeClaimTemplates` here. But How would tou fix the issue of users choosing Deployment in kind and then using `volumeClaimTemplates`?

Reply:
Fixed with condition

Reply:
I think still the volume is only there when the kind is Statefulset. Would we not need any storage if the kind is Deployment?

Reply:
I think now It's fixed